storage/innobase/srv/srv0start.cc

```diff
@@ -333,6 +333,14 @@ inline dberr_t trx_sys_t::reset_page(mtr_t *mtr)
 
   if (!sys_header) return err;
 
+  ut_a(mach_read_from_4(sys_header->page.frame + TRX_SYS + TRX_SYS_FSEG_HEADER) == 0);
+
+  /* Store the TRX_SYS_FSEG_HEADER page, offset */
+  fil_addr_t trx_fseg_addr{mach_read_from_4(sys_header->page.frame +
+                                            TRX_SYS + TRX_SYS_FSEG_HEADER + 4),
+                           mach_read_from_2(sys_header->page.frame +
+                                            TRX_SYS + TRX_SYS_FSEG_HEADER + 8)};
```

Comment:
Could we just allocate `char fseg_addr[6]` and `memcpy(fseg_addr, sys_header->page.frame + …, 6);` to it? We do not seem to need the decoded `trx_fseg_addr` anywhere; we’re only writing it back to the page.